compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/ConversionUtils.kt

```diff
@@ -40,56 +43,83 @@ internal fun <T : IrElement> FirElement.convertWithOffsets(
+fun FirTypeRef.toIrType(session: FirSession, declarationStorage: Fir2IrDeclarationStorage, irBuiltIns: IrBuiltIns): IrType {
     if (this !is FirResolvedTypeRef) {
         return createErrorType()
     }
-    return type.toIrType(session, declarationStorage)
+    return type.toIrType(session, declarationStorage, irBuiltIns)
 }
 
-fun ConeKotlinType.toIrType(session: FirSession, declarationStorage: Fir2IrDeclarationStorage, definitelyNotNull: Boolean = false): IrType {
+fun ConeKotlinType.toIrType(
+    session: FirSession,
+    declarationStorage: Fir2IrDeclarationStorage,
+    irBuiltIns: IrBuiltIns,
+    definitelyNotNull: Boolean = false
+): IrType {
     return when (this) {
         is ConeKotlinErrorType -> createErrorType()
         is ConeLookupTagBasedType -> {
-            val firSymbol = this.lookupTag.toSymbol(session) ?: return createErrorType()
-            val irSymbol = firSymbol.toIrSymbol(session, declarationStorage)
+            var irSymbol = getPrimitiveArrayType(this.classId, irBuiltIns)
+            if (irSymbol == null) {
+                val firSymbol = this.lookupTag.toSymbol(session) ?: return createErrorType()
+                irSymbol = firSymbol.toIrSymbol(session, declarationStorage)
+            }
             // TODO: annotations
             IrSimpleTypeImpl(
                 irSymbol, !definitelyNotNull && this.isMarkedNullable,
-                typeArguments.map { it.toIrTypeArgument(session, declarationStorage) },
+                typeArguments.map { it.toIrTypeArgument(session, declarationStorage, irBuiltIns) },
                 emptyList()
             )
         }
         is ConeFlexibleType -> {
             // TODO: yet we take more general type. Not quite sure it's Ok
-            upperBound.toIrType(session, declarationStorage, definitelyNotNull)
+            upperBound.toIrType(session, declarationStorage, irBuiltIns, definitelyNotNull)
         }
         is ConeCapturedType -> TODO()
         is ConeDefinitelyNotNullType -> {
-            original.toIrType(session, declarationStorage, definitelyNotNull = true)
+            original.toIrType(session, declarationStorage, irBuiltIns, definitelyNotNull = true)
         }
         is ConeIntersectionType -> {
             // TODO: add intersectionTypeApproximation
-            intersectedTypes.first().toIrType(session, declarationStorage, definitelyNotNull)
+            intersectedTypes.first().toIrType(session, declarationStorage, irBuiltIns, definitelyNotNull)
         }
         is ConeStubType -> createErrorType()
-        is ConeIntegerLiteralType -> getApproximatedType().toIrType(session, declarationStorage, definitelyNotNull)
+        is ConeIntegerLiteralType -> getApproximatedType().toIrType(session, declarationStorage, irBuiltIns, definitelyNotNull)
+    }
+}
+
+private fun getPrimitiveArrayType(classId: ClassId?, irBuiltIns: IrBuiltIns): IrClassifierSymbol? {
```

Comment:
IIRC, non-array builtin types have the same problem.

Reply:
Thanks, I'll look into non-array builtin types and address similar issues in a separate PR.